pkg/volume/testing/testing.go

```diff
@@ -400,9 +400,8 @@ var _ DeviceMountableVolumePlugin = &FakeVolumePlugin{}
 var _ NodeExpandableVolumePlugin = &FakeVolumePlugin{}
 
 func (plugin *FakeVolumePlugin) getFakeVolume(list *[]*FakeVolume) *FakeVolume {
-	volumeList := *list
-	if list != nil && len(volumeList) > 0 {
-		volume := volumeList[0]
+	if list != nil && len(*list) > 0 {
```

Comment:
Do we really need a pointer to a slice? Can we change the return value to also include a new slice?

Also, L418 is problematic.

Reply:
@msau42 , I think we do need the pointer to slice because of L418  (i.e. append to the slice in function only works via a pointer). So, if the slice is a valid/non-empty, then return the first element else create a new volume object -> initialize, append-to-list and return.

Reply:
What if you also return the new slice? Instead of modifying the passed in slice?

Reply:
@msau42 for in-place modification, we can do : 
func modifySlice(s []type) { //modify the in-place element}

However, if we have to append(grow size i.e. realloc), for that we need to pass the pointer to slice ( or in C++ terms type** instead of type*) : 
func reallocSlice(s *[]type) { // realloc the slice }

Reply:
I think it would look more idiomatic to return the updated slice than mutate the passed one in place via pointer.
However, looking at how the method is used, I think such change would cause some boilerplate around the invocation. I think it's fine the way it is.

Reply:
That's fine. But L418 still needs to be fixed

Reply:
@msau42 thanks for the review.  L418: *list = append(*list, volume) . I believe this statement is right. *list is of type []*FakeVolume and volume is of type *FakeVolume . what needs to be changed here ? Thanks.

Reply:
Just like earlier, nil pointer needs to be checked.

Reply:
@msau42 thanks, my bad.